File: frontend/src/lib/audit-judge-notes.ts
export function looksLikeFullAuditJudgePrompt(content: string): boolean {
  const text = (content || '').trim()
  if (!text) return false
  const lowered = text.toLowerCase()

  const hasCurrentStatuses =
    lowered.includes('supported')
    && lowered.includes('mismatch')
    && lowered.includes('insufficient_context')
    && lowered.includes('not_audited')
  const hasLegacyStatuses =
    lowered.includes('correct')
    && lowered.includes('incorrect')
    && lowered.includes('insufficient_context')
    && lowered.includes('evidence_not_found')
  const hasStatuses = hasCurrentStatuses || hasLegacyStatuses

  const hasRole =
    text.includes('候选')
    || text.includes('标准要求')
    || lowered.includes('evidence_candidate_keys')
    || (text.includes('审查') && text.includes('判断'))
    || text.includes('不得用常识')
  const hasOutput =
    lowered.includes('evidence_candidate_keys')
    || lowered.includes('"status"')
    || text.includes('严格输出')

  return hasStatuses && hasRole && (hasOutput || text.length > 400)
}

File: backend/app/audit_judge_notes.py
"""Helpers for optional audit_judge category notes (not full Judge prompts)."""
from __future__ import annotations


def looks_like_full_audit_judge_prompt(content: str) -> bool:
    """Detect legacy full Judge prompts wrongly stored as category notes.

    Matches both current statuses (supported/mismatch/not_audited) and older
    labels (correct/incorrect/evidence_not_found).
    """
    text = (content or "").strip()
    if not text:
        return False
    lowered = text.lower()

    has_current_statuses = (
        "supported" in lowered
        and "mismatch" in lowered
        and "insufficient_context" in lowered
        and "not_audited" in lowered
    )
    has_legacy_statuses = (
        "correct" in lowered
        and "incorrect" in lowered
        and "insufficient_context" in lowered
        and "evidence_not_found" in lowered
    )
    has_statuses = has_current_statuses or has_legacy_statuses

    has_role = (
        "候选" in text
        or "标准要求" in text
        or "evidence_candidate_keys" in lowered
        or ("审查" in text and "判断" in text)
        or "不得用常识" in text
    )
    has_output = (
        "evidence_candidate_keys" in lowered
        or '"status"' in lowered
        or "严格输出" in text
    )
    return has_statuses and has_role and (has_output or len(text) > 400)
